src/test/java/net/snowflake/ingest/streaming/FakeSnowflakeStreamingIngestChannel.java

```diff
@@ -0,0 +1,126 @@
+package net.snowflake.ingest.streaming;
+
+import java.util.LinkedHashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import net.snowflake.ingest.streaming.internal.ColumnProperties;
+
+/**
+ * Fake implementation of {@link SnowflakeStreamingIngestChannel} which stores state in memory.
+ * Should work as a drop in replacement for the original one. The implementation is able to keep
+ * state between calls: so inserting rows through {@link SnowflakeStreamingIngestChannel#insertRow}
+ * will update the {@link SnowflakeStreamingIngestChannel#getLatestCommittedOffsetToken()}.
+ *
+ * <p>The only method that does is not supported is {@link
+ * SnowflakeStreamingIngestChannel#getTableSchema}
+ */
+public class FakeSnowflakeStreamingIngestChannel implements SnowflakeStreamingIngestChannel {
+
+  private final String name;
+  private final String fullyQualifiedName;
+  private final String dbName;
+  private final String schemaName;
+  private final String tableName;
+  private final String fullyQualifiedTableName;
+  private boolean closed;
+  private String offsetToken;
+
+  private List<Map<String, Object>> rows = new LinkedList<>();
+
+  public FakeSnowflakeStreamingIngestChannel(
+      String name, String dbName, String schemaName, String tableName) {
+    this.name = name;
+    this.dbName = dbName;
+    this.schemaName = schemaName;
+    this.tableName = tableName;
```

Comment:
these fields seem required, can we add validation they are valid? non-null/ non-empty